cnf-certification-test/certification/suite.go

```diff
@@ -202,3 +209,33 @@ func testContainerCertificationStatusByDigest(env *provider.TestEnvironment, val
 	}
 	testhelper.AddTestResultLog("Non-compliant", failedContainers, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
+
+func testHelmVersion(env *provider.TestEnvironment) {
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+	tillerFound := false
+	clients := clientsholder.GetClientsHolder()
+	for _, tillerNamespace := range env.AllNamespaces {
+		// Get the Tiller pod in the specified namespace
+		podList, err := clients.K8sClient.CoreV1().Pods(tillerNamespace).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "app=helm,name=tiller",
+		})
+		if err != nil {
+			ginkgo.Fail(fmt.Sprintf("Error getting Tiller pod: %v\n", err))
+		}
+		if len(podList.Items) == 0 {
+			tnf.ClaimFilePrintf("Tiller pod not found in namespace %s\n", tillerNamespace)
+		} else {
+			tillerFound = true
+			tnf.ClaimFilePrintf("Tiller pod found in namespace %s\n, helm version is v2", tillerNamespace)
+			reportObject := testhelper.NewReportObject(fmt.Sprintf("Found Tiller pod in namespace %s, Helm Chart  version is v2 but needs to be v3 due to the security risks associated with Tiller", tillerNamespace), testhelper.HelmChart, false)
```

Comment:
Instead of a NewReportObject, you can use NewPodReportObject to save the report for the non compliant tiller pod. In this case looks like there is not need for compliant objects